agent/config/builder.go

```diff
@@ -1318,6 +1318,7 @@ func (b *Builder) svcTaggedAddresses(v map[string]ServiceAddress) map[string]str
 }
 
 func (b *Builder) serviceVal(v *ServiceDefinition) *structs.ServiceDefinition {
+	var validDNSre = regexp.MustCompile(`^[a-zA-Z0-9]([a-zA-Z0-9\-]{0,62}[a-zA-Z0-9])?$`)
```

Comment:
This look like a good idea to validate, in my case, I avhe quite a few services containing underscores (and this is tricky to change -  and while it is not supposed not to be valid, it works with all the DNS implementation I tested), I wonder if we could:
 - either make it configurable (aka: use this regexp by default if not set in configuration)
 - either being able to disable this check

Reply:
As I am going through fixing up tests I realized that for most of our tests we set the nodename to the test name (which contains underscores). Discussing now the correct approach 😬

Reply:
Yes, for instance, just doing a simple check on one of my DCs with https://github.com/criteo/consul-templaterb/blob/master/samples/find_all_invalid_dns_labels.json.erb 

=> 592 services being not Ok